src/dinit-env.cc

```diff
@@ -18,10 +18,15 @@ static void log_bad_env_cmd(int linenum)
 }
 
 // Read and set environment variables from a file. May throw std::bad_alloc, std::system_error.
-void read_env_file(const char *env_file_path, bool log_warnings, environment &env)
+void read_env_file(const char *env_file_path, bool log_warnings, environment &env, bool fatal)
```

Comment:
Can you give the parameter a better name? Maybe `throw_on_open_failure`

`fatal` sounds like it will cause process termination (and it's not very descriptive)